packages/core/src/data/strategies/SinglePostFetchStrategy.ts

```diff
@@ -182,6 +182,11 @@ export class SinglePostFetchStrategy<
 
 				if (expectedPostTypes.includes(postType)) {
 					const postTypeObject = getCustomPostType(postType, this.baseURL);
+
+					if (params.fullPath) {
+						return postPath === params.fullPath.replace(/\/?$/, '/');
+					}
+
```

Comment:
This is failing because it doesn't check for the current path with locale like further down  
`postPath === `/${this.locale}${singlePrefix}${currentPath}``

The patch I had:

```
const singlePrefix = params?.fullPath
	? ''
	: (postTypeObject?.single?.replace(/\/?$/, '') ?? '');

return (
    postPath === `${singlePrefix}${currentPath}` ||
    postPath === `/${this.locale}${singlePrefix}${currentPath}`
);
```
As it only needs to have the singlePrefix empty

Reply:
updated the logic to also account for locale, it will actually work in both cases

Reply:
thanks! approved @nicholasio